### ISSUE TITLE
16928 atoms, 3400G memory, still out of memory

### ISSUE THREAD

**Issue by iduygnay** [deepmodeling/abacus-develop#5019]
### Details

![image](https://github.com/user-attachments/assets/80dc7acd-fbc6-4aad-a8a0-94ad1adbee7c)
I just want to know how much memory is required for my task. I applied for 3400G memory but still oom. 
My calculation is using PBE functional, considering SOC, LCAO basis, 100 ecut, 16928 atoms for twisted structure. I know there are many different users who employ ABACUS to calculate twisted structures. I attached the settings of my task. 

### Task list for Issue attackers (only for developers)

- [ ] Reproduce the performance issue on a similar system or environment.
- [ ] Identify the specific section of the code causing the performance issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified performance issue.
- [ ] Implement the chosen solution to address the performance issue.
- [ ] Test the implemented solution to ensure it improves performance without introducing new issues.
- [ ] Optimize the solution if necessary, considering trade-offs between performance and other factors (e.g., code complexity, readability, maintainability).
- [ ] Review and incorporate any relevant feedback from users or developers.
- [ ] Merge the improved solution into the main codebase and notify the issue reporter.

**Comment by xdzhu** [comment]
I think it's too big for this kind of large system.

1. Reduce mpi process by `mpirun -np 1`, and enlarge the omp at the same time. 4*mpi cost 4 times memory more than 1*mpi.
2. Test `7au`, `8au`, `9au`, `10au` DZP `.orb` files on small systems first, check the convergence.
3. Reduce ecut. `ecut` test on small system is encouraged, you can refer to this notebook to test the reasonable `ecut` for your system https://bohrium.dp.tech/notebooks/55473361357
4. Test your memory cost of small systems  of choosen `.orb` and `ecut`, and then you can estimate the cost of your real twisted systems.

Notice: Do not use `gamma_only` because there is no Hamitonian output in this mode.

**Comment by QuantumMisaka** [comment]
@iduygnay The newest version of ABACUS is recommended

**Comment by iduygnay** [comment]
I don't think so the newest version supports noncolinear SOC, Like it won't converge.

**Comment by LiuXiaohui123321** [comment]
@iduygnay Could you please recompile ABACUS with the option “-DEBUG_INFO=ON” and add “out_alllog 1” to the INPUT file? In that case it will give some useful debugging information.

**Comment by mohanchen** [comment]
Since the system is too large, we will first solve the memory issue and then see if it fixes the issue. I will close it, reopen if needed.
